internal/security/spiffetokenprovider/init.go

```diff
@@ -392,12 +392,12 @@ func (b *Bootstrap) seedKnownSecrets(ctx context.Context, lc logger.LoggingClien
 	}
 
 	// copy known secrets redisdb from redis-bootstrapper to the requested service with serviceKey
-	secrets, err := secretClient.GetSecrets(fmt.Sprintf("/%s/%s", edgexRedisBootstrapperServiceKey, redisSecretName))
+	secrets, err := secretClient.GetSecret(fmt.Sprintf("/%s/%s", edgexRedisBootstrapperServiceKey, redisSecretName))
```

Comment:
we might need a way to get secrets from other service rather than itself: in this case, spiffetokenprovider is the service to retrieve secrets from redis-bootstrapper service. so this may not work

Reply:
@bnevis-i what's your thought to fix this?

Reply:
Are you sure its broken?  Did we actually change the way it works?

Reply:
If worse comes to worse, secretstore-setup has an alternative method in https://github.com/edgexfoundry/edgex-go/blob/main/internal/security/secretstore/password.go that it uses to seed known secrets.

Reply:
> we might need a way to get secrets from other service rather than itself: in this case, spiffetokenprovider is the service to retrieve secrets from redis-bootstrapper service. so this may not work

@jim-wang-intel , @bnevis-i , The method now is expecting a name not a path, so the leading slash needs to be removed. The name can be a compound name as this is building with the service key and the secret name i.e. `device-virtual/redisdb`. This is how spiffetokenprovider  is dealing with "others" service.

Reply:
> Are you sure it's broken? Did we actually change the way it works?

so, based on the go-mod-secrets implementation: https://github.com/edgexfoundry/go-mod-secrets/blob/main/pkg/types/config.go#L49-L51  the base path from the config would be spiffetokenprovider's base path when the secret name is passed in as argument for GetSecret() to provider.  However, in this corner case, we wanna copy from redis-bootstrapper's secret not itself....

Reply:
the assumption here is the base path now seems tightly coupled with the service's config itself, doesn't it?

Reply:
> the assumption here is the base path now seems tightly coupled with the service's config itself, doesn't it?

Yes, for normal services, but you will see the this service set the base path w/o the service-key so that it can accomplish the task of seeding other services Secret Store

Reply:
I think we are good now, I was just running it with Jim, print out the secretConfig, here is the log from spiffetokenprovider, you can see from the log, device virtual path seems to be correct too

level=INFO ts=2023-02-23T21:45:59.821882475Z app=security-spiffe-token-provider source=request.go:96 msg="successfully made request to Read Entity"

level=INFO ts=2023-02-23T21:45:59.887763882Z app=security-spiffe-token-provider source=request.go:96 msg="successfully made request to Create or update user"

level=INFO ts=2023-02-23T21:45:59.888737489Z app=security-spiffe-token-provider source=request.go:96 msg="successfully made request to List auth methods"

level=INFO ts=2023-02-23T21:45:59.890679092Z app=security-spiffe-token-provider source=request.go:96 msg="successfully made request to Create OIDC JWT role"

level=INFO ts=2023-02-23T21:45:59.955047821Z app=security-spiffe-token-provider source=request.go:96 msg="successfully made request to login user (service)"

level=DEBUG ts=2023-02-23T21:45:59.955072329Z app=security-spiffe-token-provider source=init.go:275 msg="seeding the known secrets if any..."

level=INFO ts=2023-02-23T21:45:59.955093093Z app=security-spiffe-token-provider source=init.go:387 msg="spiffe secret config: {vault edgex-vault 8200 /v1/secret/edgex  http    {X-Vault-Token hvs.CAESIKZ-ICEH-v_YyCJwRhPp0tObwH12YjNk7iz0fSf-PYlqGh4KHGh2cy5ROURFYjJwck5aN2FQR0M4ZkVSQ24wNkg} {false   0   }}"

level=DEBUG ts=2023-02-23T21:45:59.956767349Z app=security-spiffe-token-provider source=secrets.go:357 msg="Using Secrets URL of `http://edgex-vault:8200/v1/secret/edgex/security-bootstrapper-redis/redisdb`"

level=INFO ts=2023-02-23T21:45:59.956850539Z app=security-spiffe-token-provider source=secrets.go:277 msg="kick off token renewal with interval: 30m0s"

level=DEBUG ts=2023-02-23T21:45:59.957612364Z app=security-spiffe-token-provider source=secrets.go:424 msg="Using Secrets URL of `http://edgex-vault:8200/v1/secret/edgex/device-virtual/redisdb`"

level=INFO ts=2023-02-23T21:45:59.959867217Z app=security-spiffe-token-provider source=init.go:301 msg="successfully got raw token"

Reply:
yea, that makes sense now :-)